File: trunk/client/client/scripts/dorado/widget/form/custom-dropdown.js
/**
 * @author Benny Bao (mailto:benny.bao@bstek.com)
 * @component Trigger
 * @class 自定义下拉框。
 * @extends dorado.widget.DropDown
 */
dorado.widget.CustomDropDown = $extend(dorado.widget.DropDown,/** @scope dorado.widget.CustomDropDown.prototype */ {
	$className: "dorado.widget.CustomDropDown",
	
	ATTRIBUTES: /** @scope dorado.widget.CustomDropDown.prototype */ {
	
		/**
		 * 要显示在自定义下拉框中的控件。
		 * @type dorado.widget.Control
		 * @attribute writeBeforeReady
		 */
		control: {
			writeBeforeReady: true,
			innerComponent: ""
		}
	},
	
	createDropDownBox: function(editor, dropDown) {
		var box = $invokeSuper.call(this, arguments);
		box.set("control", this._control);
		return box;
	}
});

File: trunk/view/src/com/bstek/dorado/view/widget/form/trigger/CustomDropDown.java
package com.bstek.dorado.view.widget.form.trigger;

import com.bstek.dorado.annotation.ViewAttribute;
import com.bstek.dorado.annotation.ViewObject;
import com.bstek.dorado.annotation.Widget;
import com.bstek.dorado.annotation.XmlNode;
import com.bstek.dorado.annotation.XmlSubNode;
import com.bstek.dorado.view.widget.Control;
import com.bstek.dorado.view.widget.InnerElementReference;

/**
 * @author Benny Bao (mailto:benny.bao@bstek.com)
 * @since 2010-8-10
 */
@Widget(name = "CustomDropDown", category = "Trigger", dependsPackage = "base-widget")
@ViewObject(prototype = "dorado.widget.CustomDropDown", shortTypeName = "CustomDropDown")
@XmlNode(nodeName = "CustomDropDown")
public class CustomDropDown extends DropDown {
	private InnerElementReference<Control> controlRef = new InnerElementReference<Control>(
			this);

	@ViewAttribute
	@XmlSubNode(path = "*")
	public Control getControl() {
		return controlRef.get();
	}

	public void setControl(Control control) {
		controlRef.set(control);
	}
}
